src/components/list-view/list-view.tsx

```diff
@@ -1,20 +1,86 @@
 import React, { PureComponent } from "react";
-import { View, Text } from "react-native";
+import { View, FlatList } from "react-native";
 import { styles } from "./style";
+import { EventCard } from "./event-card";
+import { data } from "../../../tst/mock-data";
+import { EventCardProps } from "./event-card/event-card";
 
 export interface ListViewProps {}
 
-export interface ListViewState {}
+export interface ListViewState {
+  refreshing: boolean;
+  events: EventCardProps[];
+}
 
 export class ListView extends PureComponent<ListViewProps, ListViewState> {
   constructor(props: ListViewProps) {
     super(props);
+
+    this.initializeEvents = this.initializeEvents.bind(this);
+    this.fetchMoreEvents = this.fetchMoreEvents.bind(this);
+    this.onRefresh = this.onRefresh.bind(this);
+
+    this.initializeEvents();
   }
 
+  initializeEvents() {
+    const events = this.fetchEventsFromDB();
+    this.state = {
+      refreshing: false,
+      events: events,
+    };
+  }
+
+  fetchEventsFromDB() {
+    console.log("fetching initial data");
+    // TODO: Replace with call from DB
+    return data;
```

Comment:
`data` is referenced without being declared
```suggestion
    return undefined;
```

Reply:
data is imported. rn now im filling the list with the mock data im also using for the unit tests. I will replace it with a call from the DB which i referenced in the TODO statement.